src/wfs.c

```diff
@@ -181,6 +181,139 @@ wfs_t * wfs_init(const char * file, char * err)
+  if (fbuf == NULL) WFS_ERROR(err, "Input Error: Cannot open file");
+
+  char * wfs_buf = reg_match(fbuf, "WindFieldSimulation\\s*\\{.*\\}");
+  if (wfs_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid WindFieldSimulation block");
+
+  char * box_buf = reg_match(wfs_buf, "(\\{|\\s)Box\\s*\\{.*\\}");
+  if (box_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid Box block");
+
+  char * lx_buf = reg_match(box_buf, "(\\{|\\s)lx\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (lx_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid valid lx");
+
+  char * rx_buf = reg_match(box_buf, "(\\{|\\s)rx\\s*=\\s*[1-9]+[0-9]*(\\}|\\s)");
+  if (rx_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid rx");
+
+  char * lz_buf = reg_match(box_buf, "(\\{|\\s)lz\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (lz_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid lz");
+
+  char * ly_buf = reg_match(box_buf, "(\\{|\\s)ly\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (ly_buf == NULL) ly_buf = lz_buf; // in the 2D situation, ly refers to the interval over which to compute the y integral
+
+  char * rz_buf = reg_match(box_buf, "(\\{|\\s)rz\\s*=\\s*[1-9]+[0-9]*(\\}|\\s)");
+  if (rz_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid rz");
+
+  char * umean_buf = reg_match(wfs_buf, "(\\{|\\s)umean\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (umean_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid umean");
+
+  char * height_buf = reg_match(wfs_buf, "(\\{|\\s)height\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (height_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid height");
+
+  char * roughness_buf = reg_match(wfs_buf, "(\\{|\\s)roughness\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (roughness_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid roughness");
+
+  char * log_roughness_buf = reg_match(wfs_buf, "(\\{|\\s)log\\s*roughness\\s*=\\s*\\+?[0-9]+(\\.[0-9]*)?(\\}|\\s)");
+  if (log_roughness_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid log roughness");
+
+  char * spectra_buf = reg_match(wfs_buf, "(\\{|\\s)spectra\\s*=\\s*[A-Za-z]+(\\}|\\s)");
+  if (spectra_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid spectra");
+
+  char * conv_buf = reg_match(wfs_buf, "(\\{|\\s)conv\\s*=\\s*(0|1)(\\}|\\s)");
+  if (conv_buf == NULL) WFS_ERROR(err, "Input Error: Cannot find valid conv");
+
+  sim = (wfs_t *) malloc( sizeof(wfs_t) );
+  sim->box = (box_t *) malloc( sizeof(box_t) );
+  sim->box->lx = atof(strchr(lx_buf,'=') + 1);
+  sim->box->ly = atof(strchr(ly_buf,'=') + 1);
+  sim->box->lz = atof(strchr(lz_buf,'=') + 1);
+  r = atoi(strchr(rx_buf,'=') + 1);
+  sim->box->nx = 1;
+  for (i=0; i<r; i++) sim->box->nx = 2 * sim->box->nx;
+  r = atoi(strchr(rz_buf,'=') + 1);
+  sim->box->nz = 1;
+  for (i=0; i<r; i++) sim->box->nz = 2 * sim->box->nz;
+  N = sim->box->nx * sim->box->nz;
+  sim->dft_x = dft_init2D(sim->box->nx, sim->box->nz);
```

Comment:
We can add a function `int is_2d(wfs_t *) { return sim->box->ny == 1; }` and insert `if (is_2d(sim)) {...}` in `wfs_init()` to handle this part. Then I think we can remove `wfs_init2D`.